front/pages/api/w/[wId]/members/[userId]/assistant_visibility.ts

```diff
@@ -0,0 +1,191 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { getAgentConfiguration } from "@app/lib/api/assistant/configuration";
+import { Authenticator, getSession } from "@app/lib/auth";
+import { Membership, User } from "@app/lib/models";
+import { MemberAgentVisibility } from "@app/lib/models/assistant/agent";
+import { apiError, withLogging } from "@app/logger/withlogging";
+import { MemberAgentVisibilityType } from "@app/types/assistant/agent";
+
+export type PostMemberAssistantVisibilityResponseBody = {
+  created: boolean | null;
+  visibility: MemberAgentVisibilityType;
+};
+
+export const PostMemberAssistantVisibilityRequestBodySchema = t.type({
+  assistantSid: t.string,
+  visibility: t.union([
+    t.literal("workspace-unlisted"),
+    t.literal("published-listed"),
+  ]),
+});
+
+export const DeleteMemberAssistantsVisibilityBodySchema = t.type({
+  assistantSid: t.string,
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    PostMemberAssistantVisibilityResponseBody | { success: boolean }
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  const userId = parseInt(req.query.userId as string);
```

Comment:
I think we don't want to have userId being passed here (let alone using a numerical SQL id).

I know we're doing it already for members (which is not great IMHO). Maybe we can do `api/w/[wId]/members/me/assistant_visibility` ?